ColorzCore/Parser/Macros/IsDefined.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using ColorzCore.Lexer;
+
+namespace ColorzCore.Parser.Macros
+{
+    class IsDefined : BuiltInMacro
+    {
+        public EAParser ParentParser { get; private set; }
+
+        public IsDefined(EAParser parent)
+        {
+            ParentParser = parent;
+        }
+
+        public override IEnumerable<Token> ApplyMacro(Token head, IList<IList<Token>> parameters)
+        {
+            if (parameters[0].Count != 1)
+            {
+                // TODO: err somehow
```

Comment:
Probably do some param checking in the invocation of built in macros akin to how it is for preproc defs.

(you don't need to address this comment if you don't want to; I intend to)